app/app.go

```diff
@@ -732,17 +732,28 @@ func getNetworkInfo(app *App, ctx sdk.Context, manifest *UpgradeManifest) (*Netw
 	var networkInfo *NetworkConfig
 	var err error
 	if app.cudosMigrationConfigPath != "" {
-		app.Logger().Info("Loading network config", "file", app.cudosMigrationConfigPath, "hash", app.cudosMigrationConfigSha256)
+		app.Logger().Info("cudos merge: loading network config", "file", app.cudosMigrationConfigPath, "hash", app.cudosMigrationConfigSha256)
 		networkInfo, err = LoadNetworkConfigFromFile(app.cudosMigrationConfigPath, &app.cudosMigrationConfigSha256)
-		manifest.ConfigSource = fmt.Sprintf("file %s %s", app.cudosMigrationConfigPath, app.cudosMigrationConfigSha256)
 		if err != nil {
 			return nil, err
 		}
+
+		if networkInfo.ChainID != ctx.ChainID() {
+			return nil, fmt.Errorf("source file chain id %s and current node chain id %s is different", networkInfo.ChainID, ctx.ChainID())
+		}
```

Comment:
```suggestion
		if networkInfo.MergeSourceChainID != mergeSourceGenesisDoc.ChainID {
			return nil, fmt.Errorf("source file chain id %s and current node chain id %s is different", networkInfo.ChainID, ctx.ChainID())
		}
		if networkInfo.DestinationChainID != ctx.ChainID() {
			return nil, fmt.Errorf("source file chain id %s and current node chain id %s is different", networkInfo.ChainID, ctx.ChainID())
		}
```
See my [previous post](https://github.com/fetchai/fetchd/pull/383/files/959f5dd25a6f9ef8fd5a512405ee79c400879fe0..615a29b4b604b9aecbbff8ceae440357058b5251#r1787894379) .